### PR TITLE
[repo/Http] Prepare to .NET9

### PR DESCRIPTION
**Pull request by Kielek**
## Changes

Follow up #2251

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* ~~[ ] Unit tests added/updated~~
* ~~[ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes~~
* ~~[ ] Changes in public API reviewed (if applicable)~~

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.Http/HttpClientInstrumentation.cs [hunk 1]
```diff
@@ -43,14 +43,7 @@ public HttpClientInstrumentation(HttpClientTraceInstrumentationOptions options)
...
         // so that the sampler's decision is respected.
-        if (HttpHandlerDiagnosticListener.IsNet7OrGreater)
-        {
-            this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabledNet7OrGreater, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent);
-        }
-        else
-        {
-            this.diagnosticSourceSubscriber = new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabled, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent);
-        }
+        this.diagnosticSourceSubscriber = HttpHandlerDiagnosticListener.IsNet7OrGreater ? new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabledNet7OrGreater, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent) : new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabled, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent);
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Instrumentation.Http/HttpClientInstrumentation.cs:46]: nit: IMO these conditional statements are best as multi-liners. But this one is super long and hard to read 🤣

```suggestion
        this.diagnosticSourceSubscriber = HttpHandlerDiagnosticListener.IsNet7OrGreater
                ? new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabledNet7OrGreater, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent)
                : new DiagnosticSourceSubscriber(new HttpHandlerDiagnosticListener(options), this.isEnabled, HttpInstrumentationEventSource.Log.UnknownErrorProcessingEvent);
```

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs [hunk 2]
```diff
@@ -563,11 +553,11 @@ private static bool PrepareHttpWebResponseReflectionObjects(Assembly systemNetHt
 
     private static void PerformInjection()
     {
-        FieldInfo servicePointTableField = typeof(ServicePointManager).GetField("s_ServicePointTable", BindingFlags.Static | BindingFlags.NonPublic)
-            ?? throw new InvalidOperationException("Unable to access the ServicePointTable field");
+        var servicePointTableField = typeof(ServicePointManager).GetField("s_ServicePointTable", BindingFlags.Static | BindingFlags.NonPublic)
+                                     ?? throw new InvalidOperationException("Unable to access the ServicePointTable field");
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:557]: Did dotnet format add this blank space? Interesting 🤔

**Review by CodeBlanch — APPROVED** [review]
LGTM

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs (hunk 2 above)
Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:557]: For sure, it is not complaining :)

It seems that it is alliggned to typeof from the previous line.

**Pull request merged**